### ISSUE TITLE
Policy tweaks for Courier's webmail

### ISSUE THREAD

**Issue by svarshavchik** [fedora-selinux/selinux-policy#290]
The contributed courier package is missing a couple of permissions for courier's webmail cgi binary.

I was able to get the cgi binary working on F30 with the following local policy:

policy_module(courier_webmail, 1.0)

require {
	type httpd_sys_script_t;
	type courier_spool_t;
};

allow httpd_sys_script_t courier_spool_t:dir search_dir_perms;
allow httpd_sys_script_t courier_spool_t:sock_file manage_sock_file_perms;

**Comment by wrabcak** [comment]
@Koncpa, Could you please take a look on this? 

Thanks,
Lukas.

**Comment by svarshavchik** [comment]
I revisited this issue. Not sure what happened, but there in the current F31 there's another AVC getting logged:

type=AVC msg=audit(1589086763.118:1319): avc:  denied  { connectto } for  pid=674413 comm="webmail" path="/var/spool/courier/sqwebmail.sock" scontext=system_u:system_r:httpd_sys_script_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=unix_stream_socket permissive=0

I found that I had to add one more:

allow httpd_sys_script_t unconfined_service_t:unix_stream_socket connectto;

Doesn't seem ideal, but my domain knowledge in this area is limited.

**Comment by Koncpa** [comment]
Hi @svarshavchik,
we take a look on next issue, which you mentioned.

Patrik

**Comment by Koncpa** [comment]
Hi @svarshavchik,
I made pull request for first mentioned issue and I want to ask to second issue about which you write.
How do you start courier service? Via systemctl or do you start service manually?
Because when you start services manually, this service get label unconfined_service_t and then can arise issue with some permission.

Thanks,
Patrik

**Comment by svarshavchik** [comment]
Yes, some of the rpm subpackages attempt to stop and/or restart daemon services directly and not via systemctl.

However this is somewhat less important now. With RHEL6 going EOL later this year, I'll update the rpm packages to use triggers, that were implemented in rpm 4.10 from RHEL7 and up; and use systemctl in the updated packaging.

**Comment by Koncpa** [comment]
> 
> 
> Yes, some of the rpm subpackages attempt to stop and/or restart daemon services directly and not via systemctl.
> 
> However this is somewhat less important now. With RHEL6 going EOL later this year, I'll update the rpm packages to use triggers, that were implemented in rpm 4.10 from RHEL7 and up; and use systemctl in the updated packaging.

So that's the reason, why **"/var/spool/courier/sqwebmail.sock** had label **unconfined_service_t** and then arise issue with permission. Only services which starts via systemctl get right labels. 

Patrik

**Comment by Koncpa** [comment]
Hi @svarshavchik, 
so the issue is closed,  in policy is solved via boolean. By default courier cannot manage spool socket, but
when turn on boolean "# setsebool courier_manage_spool_sockets on"  it's allowed.

**Comment by svarshavchik** [comment]
Well, that's the only way it works, the webmail server expects to be able to open the socket and listen on it. So, this option will always need to be manually turned on. I suppose I can have the rpm's %post do it.
